doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,428 @@
+
+  .. insertprodn numeral digit
+
+  .. prodn::
+     numeral ::= {+ @digit } {? . {+ @digit } } {? {| e | E } {? {| + | - } } {+ @digit } }
+     int ::= {? - } {+ @digit }
+     num ::= {+ @digit }
+     digit ::= 0 .. 9
+
+Strings
+  Strings begin and end with ``"`` (double quote).  Use ``""`` to
+  represent a double quote character within a string.  In the grammar,
+  strings are identified with :production:`string`.
+
+Keywords
+  The following character sequences are reserved keywords that cannot
+  be used as identifiers::
+
+    _ Axiom CoFixpoint Definition Fixpoint Hypothesis IF Parameter Prop
+    SProp Set Theorem Type Variable as at by cofix discriminated else
+    end exists exists2 fix for forall fun if in lazymatch let match
+    multimatch return then using where with
+
+  Note that notations and plugins may define additional keywords.
+
+Other tokens
+  The set of tokens defined at any given time can vary because the
+  :cmd:`Notation` command can define new tokens.  A :cmd:`Require`
+  command may load more notation definitions, while the end of a
+  :cmd:`Section` may remove notations.  Some notations are defined in
+  the standard library (see :ref:`thecoqlibrary`) and are generally
+  loaded automatically at startup time.
+
+  Here are the character sequences that Coq directly defines as tokens
+  without using :cmd:`Notation`::
+
+    ! #[ % & ' ( () (bfs) (dfs) ) * ** + , - ->
+    . .( .. ... / : ::= := :> :>> ; < <+ <- <:
+    <<: <= = => > >-> >= ? @ @{ [ [= ] _
+    `( `{ { {| | |- || }
+
+  When multiple tokens match the beginning of a sequence of
+  characters, the longest matching token is used.  Occasionally you
+  may need to insert spaces to separate tokens.  For example, if ``~``
+  and ``~~`` are both defined as tokens, the inputs ``~ ~`` and ``~~``
+  generate different tokens, whereas if `~~` is not defined, then the
+  two inputs are equivalent.
+
+Basic notions
+-------------
+
+Glossary
+~~~~~~~~
+
+.. glossary::
+
+   term
+
+     Terms are syntactically valid objects constructed from the
+     following grammar.  When they are :term:`well-typed`, they
+     represent mathematical objects.
+
+     .. insertprodn term field_def
+
+     .. prodn::
+        term ::= @term_product_or_fun
+        | @term_let
+        | @term_if
+        | @term_fix
+        | @term_cofix
+        | @term100
+        term100 ::= @term_cast
+        | @term10
+        term10 ::= @term_application
+        | @one_term
+        one_term ::= @ @qualid_annotated
+        | @term1
+        term1 ::= @term_projection
+        | @term0 % @scope
+        | @term0
+        term0 ::= @qualid_annotated
+        | @sort
+        | @primitive_notations
+        | @term_evar
+        | @term_match
+        | @term_record
+        | @term_generalizing
+        | @term_ltac
+        | ( @term )
+        qualid_annotated ::= @qualid {? @univ_annot }
+
+     .. note::
+
+        Many commands and tactics use :n:`@one_term` rather than
```

Comment:
Why not `@term1`?

Reply:
Not sure I understand your question, but `one_term` additionally supports `@foo`. Note that `one_term` corresponds (IIRC) to `constr` in the implementation, while `term` corresponds to `lconstr`.

Reply:
Sorry, I guess my question is actually "why are one_term and term1 different nonterminals? What would go wrong of we inlined term1 into one_term and then did s/one_term/term1/ everywhere?"

Reply:
Oh! That's an interesting consideration. The syntax that is shown here is simplified (in truth `term10` is not defined in terms of `one_term`) but maybe that could be done.

Reply:
The problem is that there is an intermediate precedence level (that is not shown in the doc) that actually distinguishes further the two.

Reply:
We have:

```
term10: [
| term1 LIST1 arg
| "@" qualid OPT univ_annot LIST0 term1
| term1
]

one_term: [
| term1
| "@" qualid OPT univ_annot
]

term1: [
| term_projection
| term0 "%" scope_key
| term0
]
```

Combining one_term with one or the other would change the accepted language.  Perhaps the mlg grammar could be simplified.

Reply:
The change I did of inserting `one_term` into the `term` grammar is equivalent from the perspective of what we display in the documentation, but not in terms of the parser because there are intermediate levels that we don't cover. It will be easier to check when I push the changes to `common.edit_mlg`.

Reply:
Done.